zio-http/src/main/scala/zhttp/http/URL.scala

```diff
@@ -67,6 +67,30 @@ object URL {
     path <- Option(uri.getRawPath)
   } yield URL(Path(path), Location.Relative, queryParams(uri.getRawQuery), Fragment.fromURI(uri))
 
+  private def fromAbsoluteURI2(uri: URI): URL = {
+
+    def portFromScheme(scheme: Scheme): Int = scheme match {
+      case Scheme.HTTP  => 80
+      case Scheme.HTTPS => 443
+      case null         => -1
+    }
+
+    val scheme = Scheme.fromString2(uri.getScheme)
+    val port   = if (uri.getPort != -1) uri.getPort else portFromScheme(scheme)
```

Comment:
This code is also not safe since portFromScheme can alo return -1

Reply:
We are handling that in the next statement